scripts/actions/spells/trust/tenzen_ii.lua

```diff
@@ -12,7 +12,28 @@ spellObject.onSpellCast = function(caster, target, spell)
 end
 
 spellObject.onMobSpawn = function(mob)
-    xi.trust.message(mob, xi.trust.messageOffset.SPAWN)
+    xi.trust.teamworkMessage(mob, {
+        [xi.magic.spell.PRISHE] = xi.trust.messageOffset.TEAMWORK_1,
+    })
+
+    mob:addListener('WEAPONSKILL_USE', 'TENZEN_II_WEAPONSKILL_USE', function(mobArg, target, wsid, tp, action)
+        if wsid == 1397 then -- Oisoya
+            -- Epehemeral, fleeting, fading. You are but a memory
+            xi.trust.message(mobArg, xi.trust.messageOffset.SPECIAL_MOVE_1)
+        end
+    end)
+
+    -- Ranged Attack as much as possible (limited by 'weapon' delay)
+    mob:addSimpleGambit(ai.t.TARGET, ai.c.ALWAYS, 0, ai.r.RATTACK, 0, 0)
+
+    mob:setAutoAttackEnabled(false)
+
+    -- Gets 250 TP per hit
+    mob:addMod(xi.mod.STORETP, 210)
```

Comment:
I forget how I implemented this, but you might be able to avoid pushing this up so high by modifying his weapon stats in mob pools, which are controlled by:
```sql
  `cmbSkill` tinyint(2) unsigned NOT NULL DEFAULT 1,
  `cmbDelay` smallint(3) unsigned NOT NULL DEFAULT 240,
  `cmbDmgMult` smallint(4) unsigned NOT NULL DEFAULT 100,
```

They're still the defaults of: `3,240,100`

You'll also want to make sure these are updated on your other trust PRs. You can see Excenmille having his delay and damage boosted: `8,390,150`, and the opposite for Nanaa Mihgo: `2,190,60`

Reply:
Looks like I did this to tune Joachim's pebble throwing: `3,500,25,`

Reply:
Tenzen acts at 450 delay, and I would think should have the standard samurai traits to whatever degree is required after that. I thought this seemed odd last night but did not think to check his delay in `mob_pools`

Reply:
Zach, having played around with these values for several hours, i was never able to get it quite right.
Increasing the weapon delay in the DB does infact increase the TP return, but at the cost of 20s between each arrow being shot. Even when explicitly defining the delay in the ranged attack gambit..

Reply:
You can't explicitly define the delay for attacks in Gambits? Gambits are just bags of instructions that have no bearing on what the output action does.

See here: the "delay" is just the minimum retry time for checking the gambit again:
https://github.com/LandSandBoat/server/blob/base/src/map/ai/helpers/gambits_container.cpp#L251

This is the mechanism that Kupipi uses to not spam her debuffs if an enemy is immune, since we only have the facility to check if a target has/has_not the status:
https://github.com/LandSandBoat/server/blob/base/scripts/actions/spells/trust/kupipi.lua#L51-L54

What I was getting at here was that adding TWO HUNDRED AND TEN STORE TP as a mod was not the answer. The answer is: fiddle with Tenzen's damage and delay, and then make up the difference with some STP.

Reply:
https://github.com/LandSandBoat/server/assets/65316697/ee182584-2696-4cac-8015-c38034a2751d

Reply:
![image](https://github.com/LandSandBoat/server/assets/65316697/c5ee0550-55ad-45c8-9dce-ce9c147bbafa)

Reply:
I've removed my hold on this and pushed a new commit. Tenzen isn't perfect but he's not overpowered, and individual servers can adjust him to their taste till we get a better solution in place. He's like a gimped version of Semih at the moment, which is also how SE treats him form looking at his lower lv TP gains :(